contracts/CurrencyNetworkRegistry.sol

```diff
@@ -0,0 +1,95 @@
+pragma solidity ^0.5.8;
+
+import "./CurrencyNetworkInterface.sol";
+import "./lib/ERC165Query.sol";
+
+
+contract CurrencyNetworkRegistry is ERC165Query {
+
+    struct CurrencyNetworkMetadata {
+        address author;
+        string name;
+        string symbol;
+        uint8 decimals;
+    }
+
+    mapping (address => CurrencyNetworkMetadata) internal currencyNetworks;
+    address[] internal registeredCurrencyNetworks;
+
+    event CurrencyNetworkAdded(
+        address indexed _address,
+        address _author,
```

Comment:
You might as well index `_author` and `_name` to potentially make filtering easier without having to go through all the events since you can have three indexed arguments and we do not mind the wasted gas since we will not register a lot.

Reply:
I added _registeredBy to the indexed parameters.
Indexed string parameters seem to be not reliably decodeable with web3.py because they are basically arrays and can be split between topics and data.